filters/fadein/fadein.go

```diff
@@ -228,13 +238,16 @@ func (p *postProcessor) Do(r []*routing.Route) []*routing.Route {
 		for i := range ri.LBEndpoints {
 			ep := &ri.LBEndpoints[i]
 
+			p.endpointRegistry.SetFadeInParams(ep.Host, ri.LBFadeInDuration, ri.LBFadeInExponent)
+
 			key := endpointKey(ep.Scheme, ep.Host)
 			detected := p.detected[key].when
 			if detected.IsZero() || endpointsCreated[key].After(detected) {
 				detected = now
 			}
 
 			ep.Detected = detected
+			p.endpointRegistry.SetDetectedTime(ep.Host, detected)
```

Comment:
I think this should check if stored value is zero or endpointsCreated is after it (to override).